src/providers/node/mod.rs

```diff
@@ -580,16 +571,19 @@ fn version_number_to_pkg(version: u32) -> String {
     }
 }
 
-fn parse_regex_into_pkg(re: &Regex, node_version: &str) -> Option<String> {
-    let matches: Vec<_> = re.captures_iter(node_version).collect();
-    if let Some(captures) = matches.get(0) {
-        match captures[1].parse::<u32>() {
-            Ok(version) => return Some(version_number_to_pkg(version)),
-            Err(_e) => {}
+fn parse_node_version_into_pkg(node_version: &str) -> String {
+    let range: Range = node_version.parse().unwrap_or_else(|_| {
+        println!("Warning: node version {node_version} is not valid, using default node version {DEFAULT_NODE_PKG_NAME}");
+        Range::parse(DEFAULT_NODE_VERSION.to_string()).unwrap()
+    });
+    for version_number in AVAILABLE_NODE_VERSIONS {
```

Comment:
Should this start on the newest node version first so that `>=16` resolves to `18` instead of `16`

Reply:
done